src/wallet/walletutil.h

```diff
@@ -11,4 +11,40 @@
 //! Get the path of the wallet directory.
 fs::path GetWalletDir();
 
+//! The WalletInfo class provides wallet information.
+class WalletInfo final
+{
+    std::string m_filename;
+    bool m_external{false};
+    std::string m_name;
+    fs::path m_path;
+
+public:
+    explicit WalletInfo() {}
+    explicit WalletInfo(const std::string& filename);
+    WalletInfo(const WalletInfo& other)
+        : m_filename(other.m_filename)
+        , m_external(other.m_external)
+        , m_name(other.m_name)
+        , m_path(other.m_path) {}
+
+    //! Returns the wallet filename.
+    std::string GetFilename() const { return m_filename; };
+
+    //! Returns the wallet display name.
+    std::string GetDisplayName() const;
+
+    //! Get wallet name.
+    const std::string& GetName() const { return m_name; }
+
+    //! Get wallet absolute path.
+    const fs::path& GetPath() const { return m_path; }
+
+    //! Return whether the wallet lives in the wallet directory. See ::GetWalletDir().
+    bool IsExternal() const { return m_external; }
```

Comment:
`IsExternal()` and `m_external` seems unused?

Reply:
Now it's used, I was going to push it in another PR.